benchexec/runexecutor.py

```diff
@@ -328,7 +330,7 @@ def _init_cgroups(self):
         """
         This function initializes the cgroups for the limitations and measurements.
         """
-        self.cgroups = find_my_cgroups()
+        self.cgroups = Cgroups.from_system()
```

Comment:
This call (because it uses the `initial_cgroup=False` default value) will always allocate a new cgroup and move the current process into it.

First, I think this is not good API because a call like `Cgroups.from_system()` looks as if it just creates a `Cgroups` object, it should not have side effects. Second, I think this is likely connected to the problem of unwanted nesting as discussed in https://github.com/sosy-lab/benchexec/pull/791#issuecomment-1061992192, because the nested cgroups are named `runexec_main_` so they result from a call to `create_fresh_child_cgroup()` with `move_to_child=True`, and this is the only case where this happens except for all callers of `move_to_scope()`.

I guess the only meaningful way to handle cgroup creation in `RunExecutor` is allow controlling it from the outside. `RunExecutor` cannot know whether it should create a child cgroup and move into it, or whether the current process was already moved into a child and we should use our current parent cgroup. We could handle this by adding a parameter with the `Cgroups` instance to `RunExecutor`. If given, we should use this instance and not do anything with the current process. BenchExec (and the main method of `runexec` could then prepare this). If the instance is not given, we can for backwards compatibility create it and move the process if necessary. How does that sound?